### PR TITLE
Integrate Mayhem

### PR DESCRIPTION
**Pull request by rnshah9**

### CODE REVIEW

File: mayhem/FuzzedDataProvider.h [hunk 1]
```diff
@@ -0,0 +1,397 @@
+//===- FuzzedDataProvider.h - Utility header for fuzz targets ---*- C++ -* ===//
```
Comment (xansec) [inline mayhem/FuzzedDataProvider.h]: Please do not include FDP. This is already a part of LLVM.

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

**Comment by rnshah9** [comment]
The build does not work on versions of Ubuntu past 18.04, but the version of LLVM installed in 18.04 lacks FuzzedDataProvider.h.

**Comment by xansec** [comment]
What if we use the latest LLVM for Ubuntu 18.06? You could either add the ppa to apt, or download directly. For example:

```
diff --git a/mayhem/Dockerfile b/mayhem/Dockerfile
index 3ea5718..2bfd1d1 100644
--- a/mayhem/Dockerfile
+++ b/mayhem/Dockerfile
@@ -1,7 +1,11 @@
 FROM --platform=linux/amd64 ubuntu:18.04 as builder
 
 RUN apt-get update
-RUN DEBIAN_FRONTEND=noninteractive apt-get install -y build-essential cmake clang
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y build-essential cmake wget 
+
+RUN wget https://github.com/llvm/llvm-project/releases/download/llvmorg-15.0.6/clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04.tar.xz
+RUN tar xvf clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04.tar.xz
+ENV PATH /clang+llvm-15.0.6-x86_64-linux-gnu-ubuntu-18.04/bin:$PATH
 
 COPY . /repo
 WORKDIR /repo/build

```

**Comment by rnshah9** [comment]
I realized I could just install a later version of clang from the default repo.

**Review by xansec — APPROVED** [review]
LGTM!

**Pull request merged**